Frontend/src/pages/WeightsPage.jsx

```diff
@@ -3,250 +3,319 @@ import "../styles/classforge.css";
 import React, { useState } from "react";
 
 export default function WeightsPage() {
-    const navigate = useNavigate();
-    const [academicWeight, setAcademicWeight] = useState(50);
-    const [socialWeight, setSocialWeight] = useState(50);
-    const [wellBeingWeight, setWellBeingWeight] = useState(50);
-    const [allocationMessage, setAllocationMessage] = useState("");
-    const [datasetFile, setDatasetFile] = useState(null);
-    const [uploadMessage, setUploadMessage] = useState("");
-    const [collectionName, setCollectionName] = useState("");
-    const [isUploading, setIsUploading] = useState(false);
-     
-    // Handle file selection for dataset upload
-    const handleFileChange = (e) => {
-      if (e.target.files && e.target.files[0]) {
-        setDatasetFile(e.target.files[0]);
-      }
-    };
-  
-    // Upload dataset file to the backend
-    const handleDatasetUpload = async (e) => {
-      e.preventDefault();
-      if (!collectionName) {
-        setUploadMessage("Please enter a collection name.");
-        return;
-      }
-      
-      if (!datasetFile) {
-        setUploadMessage("Please select a dataset file.");
+  const navigate = useNavigate();
+
+  // --- primary weights (0–100) ---
+  const [academicWeight, setAcademicWeight] = useState(50);
+  const [socialWeight, setSocialWeight] = useState(50);
+  const [wellBeingWeight, setWellBeingWeight] = useState(50);
```

Comment:
Could I clarify what wellbeing weigh actually refers to in our context? Back of my mind I was just thinking it was just a placeholder for social ties, but I could be wrong? 

Same with social weight, does that actually mean anything in the backend?